src/Microsoft.AspNetCore.Razor.Evolution/Legacy/Tokenizer.cs

```diff
@@ -215,12 +215,17 @@ protected TSymbol EndSymbol(SourceLocation start, TSymbolType type)
                     errors[i] = CurrentErrors[i];
                 }
 
-                sym = CreateSymbol(start, Buffer.ToString(), type, errors);
+                sym = CreateSymbol(start, GetSymbolContent(type), type, errors);
             }
             StartSymbol();
             return sym;
         }
 
+        protected virtual string GetSymbolContent(TSymbolType type)
+        {
+            return Buffer.ToString();
+        }
```

Comment:
Are there similarly worthwhile optimizations for C#? Granted there's typically a lot more HTML than CSharp